model/Check/System/CertificateCheck.php

```diff
@@ -0,0 +1,141 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoSystemStatus\model\Check\System;
+
+use DateTime;
+use oat\oatbox\reporting\Report;
+use oat\taoSystemStatus\model\Check\AbstractCheck;
+use oat\oatbox\log\loggerawaretrait;
+
+/**
+ * Class CertificateCheck
+ * @package oat\taoSystemStatus\model\Check\System
+ * @author Aleksej Tikhanovich, <aleksej@taotesting.com>
+ */
+class CertificateCheck extends AbstractCheck
+{
+    use LoggerAwareTrait;
+
+    private const WARNING_DAYS = 14;
+    private const ERROR_DAYS = 7;
+
+    /**
+     * @inheritdoc
+     */
+    protected function doCheck(): Report
+    {
+        $certInfo = $this->getCertInfo();
+
+        if (!$certInfo) {
+            return Report::createWarning(__('Cannot get cert Info'));
+        }
+
+        $now = new DateTime();
+        $validTo = new DateTime('@' . $certInfo['validTo_time_t']);;
+        $diffDays = $validTo->diff($now)->format("%a");
+        $date = $validTo->format('Y-m-d H:i:s');
+
+        $report = Report::createSuccess($date);
+        if ($diffDays < self::ERROR_DAYS) {
+            $report = Report::createError($date);
+        } elseif ($diffDays < self::WARNING_DAYS) {
+            $report = Report::createWarning($date);
+        }
+        return $report;
+    }
+
+    /**
+     * @return array|null
+     */
+    private function getCertInfo() : ?array
+    {
+        if (!$url = \tao_helpers_Uri::getRootUrl()) {
+            return null;
+        }
+        $certInfo = null;
+
+        $originalParse = parse_url($url, PHP_URL_HOST);
+        $get = stream_context_create(
+            [
+                "ssl" => [
+                    "capture_peer_cert" => TRUE
+                ]
+            ]
+        );
+
+        if ($read = stream_socket_client(
+            sprintf(
+                'ssl://%s:443',
+                $originalParse
+            ),
+            $errno,
+            $errStr,
+            30,
+            STREAM_CLIENT_CONNECT,
+            $get
+        )) {
+            $cert = stream_context_get_params($read);
+            if (isset($cert['options']['ssl']['peer_certificate'])) {
+                $certInfo = openssl_x509_parse($cert['options']['ssl']['peer_certificate']);
+            }
+        }
+        return $certInfo;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isActive(): bool
+    {
+        if ((!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off')
+            || $_SERVER['SERVER_PORT'] == 443) {
+            return true;
```

Comment:
More a comment, as I'm not sure (and not 100% sure how to check it on local dev env):
Are you sure that all `$_SERVER` stuff would be available under CLI? Because if I'm not mistaken, you can only get `$_SERVER['HTTPS']` when working with request, not with CLI.
Could you please check it?

Reply:
true